Docs: Fix TestData docs (#22279)
grafana/grafana: docs/sources/features/datasources/testdata.md

+++

# Grafana TestData DB

The purpose of this data source is to make it easier to create fake data for any panel.
Using `TestData DB` you can build your own time series and have any panel render it.
This make is much easier to verify functionally since the data can be shared very easily.

## Enable

> The `TestData DB` data source is not enabled by default. To enable it, first navigate to the *Data Source* section of the configuration menu. Click the *Add Data Source*, search for and select `TestData DB`. Finally, click *Save & Test* to enable it.

Reviewer: Per the [Documentation style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md) and [Microsoft Style Guide](https://docs.microsoft.com/en-us/style-guide/welcome/), clickable items in the UI should be bold, not italic. Please change that. 

Also, step-by-step instructions should be in numbered step format. This will make them easier to read.

By the way, did you test whether you needed to enable the TestData DB data source? I have played with it a little, and it was always available by default.

Author: > clickable items in the UI should be bold, not italic

Done

> step-by-step instructions should be in numbered step format

Done

> did you test whether you needed to enable the TestData DB data source?

Yes, it appears this is required. Just tested on a brand-new installation.

Reviewer: Installation on which OS?

Author: Grafana v6.6.1 (21bf8b71bc) on Ubuntu 19.10 (installed w/ apt)

Reviewer: Okay, I figured out why I am confused. You do need to enable it...the same as every other data source. LOL.

+++

# Grafana TestData DB

The purpose of this data source is to make it easier to create fake data for any panel.
Using `TestData DB` you can build your own time series and have any panel render it.
This make is much easier to verify functionally since the data can be shared very easily.

## Enable

> The `TestData DB` data source is not enabled by default. To enable it, first navigate to the "Data Source" section of the configuration menu. 

Reviewer: ```suggestion
The `TestData DB` data source is not enabled by default. To enable it: 
1. In the **Configuration** menu (small gear on the left side of the screen, click **Data Sources**. 
```

> 1. Click the **Add Data Source**

Reviewer: ```suggestion
1. Click **Add Data Source**.
```

> 2. Search for and select `TestData DB`

Reviewer: ```suggestion
2. Search and click `TestData DB`.
```

> 3. Finally, click **Save & Test** to enable it

Reviewer: ```suggestion
3. Click **Save & Test** to enable it.
```

## Create mock data.

> Once `TestData DB` is enabled you can use it as a data source in any metric panel.

Reviewer: ```suggestion
Once `TestData DB` is enabled, you can use it as a data source in any metric panel.
```

![](/img/docs/v41/test_data_add.png)

## CSV

The comma separated values scenario is the most powerful one since it lets you create any kind of graph you like.

> Once you provided the numbers `TestData DB` will distribute them evenly based on the time range of your query.

Reviewer: ```suggestion
Once you provided the numbers, `TestData DB` distributes them evenly based on the time range of your query.
```

![](/img/docs/v41/test_data_csv_example.png)

## Dashboards

> `TestData DB` also contains some dashboards with example. Click on **Configuration** > **Data Sources** > **TestData DB** > **Dashboards**, then import the "Simple Streaming Example" dashboard.

Reviewer: ```suggestion
`TestData DB` also contains some dashboards with examples. 
1. Click **Configuration** > **Data Sources** > **TestData DB** > **Dashboards**.
1. **Import** the **Simple Streaming Example** dashboard.
```

### Commit updates to the dashboards

> If you want to submit a change to one of the current dashboards bundled with `TestData DB` you have to update the revision property.

Reviewer: ```suggestion
To submit a change to one of the current dashboards bundled with `TestData DB`, update the revision property.
```
